### PR TITLE
Add design doc for volcano extender plugin

### PR DESCRIPTION
**Pull request by xing0821**
Extender plugin design doc in (https://github.com/volcano-sh/volcano/issues/1935)

### CODE REVIEW

**Review by shinytang6 — COMMENTED** [review]
lgtm overall 👍

File: docs/design/extender.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  Implement a plugin named extender, the plugin will register related functions in the SessionOpen function, and pass the snapshot to the endpoint through the http method.
+  The extender plugin will choose which methods to register according to the configuration file, and is responsible for serializing and deserializing the parameters. When there is an error in the http call, the plugin decides how to deal with it according to the configuration file.
+### Extender Arguments
+```
+- plugins:
+   - name: extender
+     arguments:
+       extender.urlPrefix: http://127.0.0.1
+       extender.httpTimeout: 100ms
+       extender.openSessionOpenVerb: onSessionOpen
```
Comment (k82cn) [inline docs/design/extender.md:25]: prefer to have default value for those verbs to make configuration simple :)

Comment (k82cn) [reply docs/design/extender.md:25]: and have a parameter `extender.defaultVerbs = false` to disable all default verbs if necessary.

Comment (k82cn) [reply docs/design/extender.md:25]: anyway, we can improve that later; as I'm not sure which one is simpler :)

Comment (xing0821) [reply docs/design/extender.md:25]: If plugin provide default value of verbs, how to distinguish between default value and null value(null value means that the function has not been defined by the user) 
Maybe we can add an example configuration in example package

Comment (k82cn) [reply docs/design/extender.md:25]: that's ok :)

Comment (xing0821) [reply docs/design/extender.md:25]: > and have a parameter `extender.defaultVerbs = false` to disable all default verbs if necessary.

that's a good idea. We can do some user interviews to make configuration more simpler

**Pull request merged**